IdentityCore/src/MSIDError.h

```diff
@@ -39,5 +40,5 @@ typedef NS_ENUM(NSInteger, MSIDErrorCode)
     removeMeWhenThingsAreAdded = -10000,
 };
 
-extern NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError);
+extern NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError, NSDictionary *additionalUserInfo);
```

Comment:
@oldalton Do we want `correlationId` to be a separate parameter here or want it to be passed within `additionalUserInfo` here?

Reply:
I'd say separate. What do you think? @jasoncoolmax @unpluggedk @RPangrle

Reply:
I would say separate. The reason I have is, if it is not separate, we will easily forget to pass correlation id when creating error even if we have it. Then ADAL will suffer as it could find the correlation id when it should :)